twenty-first/src/shared_math/bfield_codec.rs

```diff
@@ -707,7 +699,11 @@ mod bfield_codec_tests {
     fn test_decode_random_negative() {
         for _ in 1..=10000 {
             let len = random_length(100);
-            let str: Vec<BFieldElement> = random_elements(len);
+            let mut str: Vec<BFieldElement> = random_elements(len);
+            // Claiming a length that is too large leads to error “capacity overflow” when decoding.
+            if !str.is_empty() {
+                str[0] = (100 + random_length(500) as u64).into();
+            }
```

Comment:
I'm not super happy about this. The underlying requirement comes from initiating the result vector as `Vec::with_capacity(num_elements)` in `impl<T> BFieldCodec for Vec<T>::decode()`, which is nice to have but apparently introduces a DOS vector. The alternative is to use `vec![]` and bite the cost of internal re-allocations.

Thought, @Sword-Smith?

Reply:
After writing above comment, I feel much more strongly inclined to use `vec![]` instead of `Vec::with_capacity(random_persons_claim)`.